tests/kernel/msgq/msgq_api/src/test_msgq_contexts.c

```diff
@@ -208,6 +215,89 @@ void test_msgq_isr(void)
 	msgq_isr(&kmsgq);
 }
 
+/**
+ * @see k_msgq_get()
+ */
+static void thread_entry_get_data(void *p1, void *p2, void *p3)
+{
+	u32_t rx_buf[MSGQ_LEN];
+	int i = 0;
+
+	while (k_msgq_get(p1, &rx_buf[i], K_NO_WAIT) != 0) {
+		++i;
+	}
+
+	k_sem_give(&end_sema);
+}
+
+/**
+ * @see k_msgq_put()
+ */
+static void pend_thread_entry(void *p1, void *p2, void *p3)
+{
+	int ret;
+
+	ret = k_msgq_put(p1, &data[1], TIMEOUT);
+	zassert_equal(ret, 0, NULL);
+}
+
+/**
+ * @see k_msgq_put(), k_msgq_purge()
+ */
+static void msgq_thread_data_passing(struct k_msgq *pmsgq)
+{
+
+	while (k_msgq_put(pmsgq, &data[0], K_NO_WAIT) != 0) {
+	}
+
+	k_tid_t tid = k_thread_create(&tdata2, tstack2, STACK_SIZE,
+					pend_thread_entry, pmsgq, NULL,
+					NULL, K_PRIO_PREEMPT(0), 0, 0);
+
+	k_tid_t tid1 = k_thread_create(&tdata1, tstack1, STACK_SIZE,
+					thread_entry_get_data, pmsgq, NULL,
+					NULL, K_PRIO_PREEMPT(1), 0, 0);
+
+	k_sem_take(&end_sema, K_FOREVER);
+	k_thread_abort(tid);
+	k_thread_abort(tid1);
+
+	/**TESTPOINT: msgq purge*/
+	k_msgq_purge(pmsgq);
+}
+
+/**
+ * @see k_msgq_init()
+ */
+void test_msgq_pend_thread(void)
+{
+	k_msgq_init(&msgq1, tbuffer1, MSG_SIZE, 1);
+	k_sem_init(&end_sema, 0, 1);
+
+	msgq_thread_data_passing(&msgq1);
+}
+
+/**
+ * @see k_msgq_alloc_init(), k_msgq_cleanup()
+ */
+void test_msgq_alloc(void)
```

Comment:
document the test please

Reply:
@nashif documentation done as part of #9317.